### ISSUE TITLE
Judgement of Wisdom Not Proccing From Attacks at Correct Rate

### ISSUE THREAD

**Issue by spacct** [Atlantiss/NetherwingBugtracker#9070]
[//]: # (Enclose links to things related to the bug using http://wowhead.com or any other TBC database.)
[//]: # (You can use screenshot ingame to visual the issue.)
[//]: # (Write your tickets according to the format:)
[//]: # ([Quest][Azuremyst Isle] Red Snapper - Very Tasty!)
[//]: # ([NPC] Magistrix Erona)
[//]: # ([Spell][Mage] Fireball)
[//]: # ([Npc][Drop] Ghostclaw Lynx)
[//]: # ([Web] Armory doesnt work)
[//]: # (To find server revision type `.server info` command in the game chat.)

**Description**: Judgement of Wisdom is proc'ing much less frequently than expected from physical attacks.

**Current behaviour**: Out of 375 attack hits, Judgement of wisdom restored 8,880 mana (or 120 procs), giving it a rate of 32%. Below are pictures from the combat log. These vast swaths without a JoW proc are commonplace. 

Note: this test was done while not at 100% mana to ensure that having full mana was not the cause of the bug.

I believe this bug may be indicative of a problem with many chance on hit procs (trinkets like dst, other trinkets and class talents, etc). For whatever reason, many hits are not registering with the proc chances.

![image](https://github.com/Atlantiss/NetherwingBugtracker/assets/104590778/5a9f4b99-6f5e-4b93-8e71-582659b3283c)

Pictures from Details showing the 375 attack hits (217 steady hits and 158 auto shot hits), as well as the 8,880 mana restored from these cumulative attacks.
![image](https://github.com/Atlantiss/NetherwingBugtracker/assets/104590778/b4a425bf-b9b3-423b-8d63-5dce145af4a9)
![image](https://github.com/Atlantiss/NetherwingBugtracker/assets/104590778/c5ff062e-5463-46aa-884c-52c31e80e3f7)
![image](https://github.com/Atlantiss/NetherwingBugtracker/assets/104590778/a56fcff8-0a16-4830-ae26-7c9acbb0a295)

**Expected behaviour**: Judgement of Wisdom should be proc'ing from melee and ranged attacks at a 50% rate. The 8,880 mana restored in this case should have been 13,838. This is a near 5,000 mana difference, and 5,000 mana is a huge difference on a raid boss encounter.

<img width="595" alt="image" src="https://github.com/Atlantiss/NetherwingBugtracker/assets/104590778/d8c59923-cbb1-416c-a7fd-eec35e51e6df">
https://www.wowhead.com/tbc/spell=20354/judgement-of-wisdom

**Server Revision**:

**Comment by spacct** [comment]
Upon further investigation, this bug is confirmed through raid boss combat logs. Further analysis on a larger sample was done on boss logs to confirm.

Faceless logs from Brutallus kill on Apr 28 show that after Judgement of Wisdom was applied, it stayed on the boss throughout the remainder of the fight (final 3:51 of the 4:48 fight). For those classes using mana bars (excluding pets, warriors, rogue, feral druid), 3,405 attacks were made on the boss after Judgement of Wisdom was applied (see table below), and 1,036 of those attacks resulted in a Judgement of Wisdom proc. This is a rate of 0.3043 (30.43%), and this corroborates the post above (which showed a 32% proc rate during the test of 375 attack hits).

Note: the attack analysis below excludes Seal of Blood and Seal of Command which both appear to proc Judgement of Wisdom (for the purpose of care not to include anything improperly). This makes the above 30.43% rate conservatively high (it is likely even lower).

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/premo/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/premo/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


Attack/Spell Name | Count
-- | --
Melee Swing | 673
Aimed Shot | 9
Arcane Blast | 126
Arcane Shot | 36
Auto Shot | 634
Chain Lightning | 8
Crusader Strike | 38
Death Coil | 3
Earth Shock | 53
Exorcism | 2
Fireball | 1
Flame Shock | 23
Frostbolt | 30
Judgement of Blood | 46
Lightning Bolt | 195
Mind Blast | 31
Moonfire | 19
Multi-Shot | 43
Shadow Bolt | 452
Shadowburn | 4
Starfire | 76
Steady Shot | 604
Stormstrike | 96
Windfury Attack | 203
Grand Total | 3405



</body>

</html>

**Comment by Wolffenstein** [comment]
Fixed in rev 3991.

**Comment by spacct** [comment]
Tested on PTR follow server update. Judgement of Wisdom still proc'ing at lower than expected rate. Appears to be unaffected by change to steady shot.

171 hits, 4,292 mana restored (58 procs); 33.92% proc chance.

Tested Go For the Throat hunter talent proc and Bow of the Betrayer mana drain, and after short tests, both appear to be working properly following server update.

For additional context, I used the Faceless logs mentioned in my previous comment to compare procs for Enhancement Shaman melee attacks and damaging spells, and the proc rate for the Shaman for Judgement of Wisdom was:

103 Judgement of Wisdom procs, 383 attacks hit for a 26.89% proc chance.

I believe the Judgement of Wisdom proc chance is separately bugged from the Steady Shot issue affecting Go for the Throat and Black Bow mana drain.

Combat log for reference on most recent Judgement of Wisdom test on PTR following server update (50% proc chance intended; existing proc rate is ~30%):
![image](https://github.com/Atlantiss/NetherwingBugtracker/assets/104590778/7c79e12b-19a0-48a0-a40f-19e1140a9805)
